src/core/thread/announce_sender.cpp

```diff
@@ -69,7 +69,7 @@ otError AnnounceSenderBase::SendAnnounce(Mac::ChannelMask aChannelMask,
     VerifyOrExit(aPeriod != 0, error = OT_ERROR_INVALID_ARGS);
     VerifyOrExit(aJitter < aPeriod, error = OT_ERROR_INVALID_ARGS);
 
-    aChannelMask.Intersect(Mac::ChannelMask(Phy::kSupportedChannels));
+    aChannelMask.Intersect(Mac::ChannelMask(Get<Mac::Mac>().GetSupportedChannelMask().GetMask()));
```

Comment:
I think we may be able to simplify the code like this: 
`aChannelMask.Intersect(Get<Mac::Mac>().GetSupportedChannelMask()))`